src/Middleware/CORS/src/Infrastructure/CorsMiddleware.cs

```diff
@@ -145,8 +145,23 @@ private async Task InvokeCore(HttpContext context, ICorsPolicyProvider corsPolic
             // Get the most significant CORS metadata for the endpoint
             // For backwards compatibility reasons this is then downcast to Enable/Disable metadata
             var corsMetadata = endpoint?.Metadata.GetMetadata<ICorsMetadata>();
+
             if (corsMetadata is IDisableCorsAttribute)
             {
+                var isOptionsRequest = string.Equals(
```

Comment:
Matches the current behavior of the CORS filter in MVC. The filter short-circuits if a `IDisableCorsAttribute` is present and you're doing a preflight request